editor.planx.uk/src/pages/FlowEditor/components/Team/components/SettingsModal.tsx

```diff
@@ -0,0 +1,46 @@
+import Dialog from "@mui/material/Dialog";
+import React from "react";
+
+import { EditorModalProps } from "../types";
+import { EditorUpsertModal } from "./EditorUpsertModal";
+import { RemoveUserModal } from "./RemoveUserModal";
+
+export const SettingsModal = ({
```

Comment:
I'm not sure this is the right abstraction here (though I agree on your instincts to try and share the styles etc).

This looks like a generic modal that could be used anywhere (going by the component name and location within the codebase), but it infact takes in a `EditorModalProps` and needs the user related  ``data-testid={`dialog-${actionType}-user`}``.

Here's what I think we should be aiming for - 
 - Shared modal styles
 - The ability to pass a `children` prop which is of type `EditorUpsertModal`
 
 What this would achieve is -
  - A truly generic and visually consistent `SettingsModal`
  - The ability to nest anything within in (user details, something to do wih settings, or themes etc)
  
There's a few ways to achieve this, but one of the most powerful and simple is likely via the MUI theme. It's likely there's opportunity to tidy up a few `Dialog` components by this method.

Reply:
Will do Styled Component in this PR and do a follow up refactor PR with using the theme.ts style overrides